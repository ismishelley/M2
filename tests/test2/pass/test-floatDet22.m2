int main(){
	matrix float [2][2] m;
	float det;
	m = [[1.0,2.0];
		 [3.0,4.0]];
	det = m[0][0] * m[1][1] - m[1][0] * m[0][1];

	printStr("The determinant of matrix");
	myprint(m);
	printStr("is");
	printFloat(det);
	
	return 0;
}

void myprint(matrix float [2][2] M){
  int i;
  int j;
  for(i = 0; i < M:rows; ++i){
    for(j = 0; j < M:cols; ++j){
      printFloat(M[i][j]);
    }
    printStr("");
  }
}