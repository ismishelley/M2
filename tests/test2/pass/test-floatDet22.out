The determinant of matrix
1.000000	2.000000	
3.000000	4.000000	
is
-2.000000	